wrench/src/main.rs

```diff
@@ -665,17 +700,13 @@ fn main() {
         winit::ControlFlow::Continue
     };
 
-    match events_loop {
+    match *events_loop {
         None => {
-            while body(&mut wrench, winit::Event::Awakened) == winit::ControlFlow::Continue {}
+            while body(wrench, winit::Event::Awakened) == winit::ControlFlow::Continue {}
             let rect = DeviceUintRect::new(DeviceUintPoint::zero(), size);
             let pixels = wrench.renderer.read_pixels_rgba8(rect);
             save_flipped("screenshot.png", pixels, size);
         }
-        Some(ref mut events_loop) => {
-            events_loop.run_forever(|event| body(&mut wrench, event));
-        }
+        Some(ref mut events_loop) => events_loop.run_forever(|event| body(wrench, event)),
     }
-
-    wrench.renderer.deinit();
```

Comment:
what's this part of the change about?

Reply:
The call do `deinit()` moved to the caller in order to keep it in the same function as the call to the constructor.